x-pack/plugins/maps/common/descriptor_types/map_descriptor.ts

```diff
@@ -56,11 +56,12 @@ export type TooltipState = {
 
 export type DrawState = {
```

Comment:
not sure if we want to hack `drawState` for editing geometries. This was really for drwaing filter objects, and so it has a few things in here that are not relevant. e.g. `relation`, `geoFieldType` (to construct the approprite filter), `geometry/filterLabel` (for human readable formatting), ...

when orienting to make this layer-based, we can strip this overload. We could introduce a new concept, `EditState`, and just start with that

`
{
 layerId: string;
 shapeTypeToDraw: DRAW_TYPE;
}
`